webapp/src/main/java/io/github/microcks/web/SoapController.java

```diff
@@ -179,13 +194,21 @@ public ResponseEntity<?> execute(
          Response response = null;
          DispatchContext dispatchContext = null;
 
-         // Depending on dispatcher, evaluate request with rules.
-         if (DispatchStyles.QUERY_MATCH.equals(dispatcher)) {
-            dispatchContext = getDispatchCriteriaFromXPathEval(dispatcherRules, body);
-         } else if (DispatchStyles.SCRIPT.equals(dispatcher)) {
-            dispatchContext = getDispatchCriteriaFromScriptEval(dispatcherRules, body, request);
-         } else if (DispatchStyles.RANDOM.equals(dispatcher)) {
-            dispatchContext = new DispatchContext(DispatchStyles.RANDOM, null);
+         try {
+            // Depending on dispatcher, evaluate request with rules.
+            if (DispatchStyles.QUERY_MATCH.equals(dispatcher)) {
+               dispatchContext = getDispatchCriteriaFromXPathEval(dispatcherRules, body);
+            } else if (DispatchStyles.SCRIPT.equals(dispatcher)) {
+               dispatchContext = getDispatchCriteriaFromScriptEval(dispatcherRules, body, request);
+            } else if (DispatchStyles.RANDOM.equals(dispatcher)) {
+               dispatchContext = new DispatchContext(DispatchStyles.RANDOM, null);
+            } else {
+               return new ResponseEntity<String>(
+                  String.format("The dispatch %s is not supported!", dispatcher),
+                  HttpStatus.NOT_FOUND);
+            }
+         } catch (ResponseStatusException e) {
+            return new ResponseEntity<String>(e.getMessage(), e.getStatusCode());
```

Comment:
Shouldn't we force the `ResponseEntity` status code to be 500 - Internal Error as this happens when we got a dispatcher evaluation exception?

Reply:
Yes, we can force the status code. 
In fact I would have preferred to catch the ResponseStatusException at upper level (by using an exception handler for instance) instead of using ResponseEntity to return errors. What are you thoughts about it ?

Reply:
By default, spring will handle exceptions. If you raise an exception in a controller, spring will return an 500 error automatically.

To control what spring do with our exceptions, there are differents alternatives.

The easiest way is to use ResponseStatusException to control the http status in the exception itself.
If we want to add the exception message to the error message returned by Srping, we just have to enable it in properties : 
`server.error.include-message=ALWAYS`
The drawback is that any exception message will be returned to the user. Sometimes, for security reason, we do not what that. By the way, I didn't think that it is a problem for microcks.
We could also want to log the exceptions when Spring catch them for us. This is possible again in property file : 
`spring.mvc.logResolvedException=true`

If we want to have full control of our exceptions, the best alternative is to define our own exception handler in a ControllerAdvice bean : https://docs.spring.io/spring-framework/reference/web/webmvc/mvc-controller/ann-exceptionhandler.html